app/src/main/java/io/intrepid/animationexercise/screens/detail/DetailActivity.java

```diff
@@ -101,6 +118,16 @@ public void showCat() {
     public void animatePowerUp() {
         Drawable drawable = powerUpBar.getDrawable();
         drawable.setLevel(10000);
+        ObjectAnimator moveAnim = ObjectAnimator.ofFloat(powerUpBar, "Y", 1000);
+        moveAnim.setDuration(20000);
+        moveAnim.addListener(new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                drawable.setLevel(0);
```

Comment:
I do like this idea of resetting the level back to zero, but it might be confusing to the user if the animation immediately resets itself.